### DISCUSSION TITLE
working with many to many relationships

### DISCUSSION THREAD

**Discussion by kvedak** [orchidsoftware/platform#1553] [Q&A] [answered]
I recently discovered the Orchid package. You have done a great job. In the Data management Example, you have shown how to work with one-to-many relationships. In my project, I want to use many-many relationship and am having difficulties. I have two models Career and Interset with many-to-many relationships

In Career Model I have defined Interests as 
`public function interests()
    {
        return $this->belongsToMany(interest::class, 'career_interest');
    }`
In the CareerEditSceen Layout() function I have 

`Relation::make('career.interests.interest_id.')
                    ->fromModel(Interest::class, 'title')
                    ->multiple()
                    ->title('Choose Career Interest'),`

The dropdown works fine but the data is not saved in the database. What am I doing wrong? I have also tried using 
`career.interests` in the make function. 

As you can see I am not an expert in Laravel, and need your help. 

Thanks in advance

**Comment by hellodit** [comment, marked as answer]
Have you read the Many to many documentation [here](https://laravel.com/docs/8.x/eloquent-relationships#many-to-many)? 
To save data you can use `attach` function

**Reply by kvedak** [reply]
Thanks, for the information.

**Reply by EeKay** [reply]
You can also use the `sync` function. An example on how to implement this was already given in this discussion: https://github.com/orchidsoftware/platform/discussions/1724#discussioncomment-723637

**Comment by willardinteractive** [comment]
@kvedak did you come up with a solution? I am struggling with this as well.

**Reply by hellodit** [reply]
What is your problem, maybe I can help you

**Comment by kvedak** [comment]
Sorry for not updating the conversation, Asdita. My problem was my
ignorance. I was under the impression that Orchid has built-in support for
one-to-many relationships. After your direction, I read the documentation
and updated my code with attach function.
Thanks for your help.
Kiran
